plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -0,0 +1,552 @@
+            JSONObject updateCenterJson = new JSONObject(updateCenterText);
+            return updateCenterJson;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+
+    public void resolveDependencies(Plugin plugin) {
+        JSONObject updateCenterJson = getUpdateCenterJson();
+
+        if (updateCenterJson == null) {
+            System.out.println("Unable to get update center json");
+        }
+
+        JSONObject plugins = updateCenterJson.getJSONObject("plugins");
+        JSONObject pluginInfo = (JSONObject) plugins.get(plugin.getName());
+        JSONArray dependencies = (JSONArray) pluginInfo.get("dependencies");
+
+        if (dependencies == null || dependencies.length() == 0) {
+            System.out.println(plugin.getName() + " has no dependencies");
+            return;
+        }
+
+        List<Plugin> dependentPlugins = new ArrayList<>();
+
+        System.out.println("Plugin depends on: ");
+
+        for (int i = 0; i < dependencies.length(); i++) {
+            JSONObject dependency = dependencies.getJSONObject(i);
+            String pluginName = dependency.getString("name");
+            String pluginVersion = dependency.getString("version");
+            boolean isPluginOptional = dependency.getBoolean("optional");
+            Plugin dependentPlugin = new Plugin(pluginName, pluginVersion, isPluginOptional);
+            dependentPlugins.add(dependentPlugin);
+
+            System.out.println(pluginName + ": " + pluginVersion);
+        }
+
+
+        for (Plugin dependency : dependentPlugins) {
+            String dependencyName = dependency.getName();
+            String dependencyVersion = dependency.getVersion();
+            if (dependency.getPluginOptional()) {
+                System.out.println("Skipping optional dependency " + dependencyName);
+                continue;
+            }
+
+            String installedVersion = "";
+            if (installedPluginVersions.containsKey(plugin.getName())) {
+                installedVersion = installedPluginVersions.get(dependencyName);
+            } else if (bundledPluginVersions.containsKey(plugin.getName())) {
+                installedVersion = bundledPluginVersions.get(dependencyName);
+            }
+
+            if (!StringUtils.isEmpty(installedVersion)) {
+                if (comparePluginVersions(installedVersion, dependencyVersion) < 0) {
+                    System.out.println("Installed version of " + dependencyName + " is less than minimum " +
+                            "required version of " + dependencyVersion + ", upgrading bundled dependency");
+                } else {
+                    System.out.println("Skipping already installed dependency ");
+                }
+            } else {
+                downloadPlugin(dependency);
+            }
+        }
+
+    }
+
+
+    public int comparePluginVersions(String version1, String version2) {
```

Comment:
Please use https://github.com/jenkinsci/lib-version-number/blob/master/src/main/java/hudson/util/VersionNumber.java instead

Reply:
Will be fixed in next PR. Created Jira task to track: https://issues.jenkins-ci.org/browse/JENKINS-58078